### PR TITLE
LPS-179691 Add REST endpoint to manage folder ratings

### PR DESCRIPTION
**Pull request by adolfopa**
# What is this about?

https://issues.liferay.com/browse/LPS-179691

Add an endpoint to manage folder ratings (GET, POST, PUT, DELETE).

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentFolderResourceImpl.java [hunk 1]
```diff
@@ -402,6 +447,44 @@ private Page<DocumentFolder> _getDocumentFoldersPage(
...
+							"getDocumentFolderMyRating", folder.getUserId(),
+							DLFolder.class.getName(), folder.getGroupId())
+					).put(
+						"replace",
+						addAction(
+							ActionKeys.VIEW, folder.getPrimaryKey(),
+							"putDocumentFolderMyRating", folder.getUserId(),
+							DLFolder.class.getName(), folder.getGroupId())
+					).build(),
+					_portal, ratingsEntry, _userLocalService);
```
Comment (marcogalluzzi) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentFolderResourceImpl.java:483]: Would it be useful to create a private method like this?

```java
	private Map<String, String> _addViewAction(Folder folder, String methodName) {
		return addAction(
			ActionKeys.VIEW, folder.getPrimaryKey(),
			methodName, folder.getUserId(),
			DLFolder.class.getName(), folder.getGroupId());
	}
```

**Review by marcogalluzzi — CHANGES_REQUESTED** [review]

File: modules/apps/headless/headless-delivery/headless-delivery-impl/rest-openapi.yaml [hunk 2]
```diff
@@ -7662,6 +7662,112 @@ paths:
                             schema:
                                 $ref: "#/components/schemas/Document"
             tags: ["Document"]
+    "/document-folders/{documentFolderId}/my-rating":
+        delete:
+            description:
+                Deletes the document folder's rating and returns a 204 if the operation succeeded.
```
[the remark this answers is not in the document]
Comment (adolfopa) [reply modules/apps/headless/headless-delivery/headless-delivery-impl/rest-openapi.yaml:7668]: I based this text in the existing documentation for Document ratings; that has been reviewed by Brian and the doc. team, so I think it is better to keep this consistent with that.

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentFolderResourceImpl.java (hunk 1 above)
Comment (adolfopa) [reply modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentFolderResourceImpl.java:483]: I'm following existing patterns, so I'm not sure about this; If you want to give it a try, we can send it in a different PR so we don't delay this.

**Comment by adolfopa** [comment]
Rebased and fixed conflicts.

**Comment by marcogalluzzi** [comment]
ci:forward

**Pull request closed without merging**